fieldtracing/fieldtracing.cpp

```diff
@@ -996,114 +1010,213 @@ namespace FieldTracing {
                      ) {
                         cellNeedsContinuedBWTracing[n] = 0;
                         cellBWTracingCoordinates[n] = {0,0,0};
-                        cellBWConnection[n] = TracingLineEndType::OPEN;
+                        cellBWConnection[n] += TracingLineEndType::OPEN;
+                        break;
+                     }
+                     
+                     // If we exceed the max tracing distance we're probably looping
+                     if(cellBWRunningDistance[n] > maxTracingDistance) {
+                        cellNeedsContinuedBWTracing[n] = 0;
+                        cellBWTracingCoordinates[n] = {0,0,0};
+                        cellBWConnection[n] += TracingLineEndType::LOOP;
+                        #pragma omp critical
+                        {
+                           warnMaxDistanceExceeded = true;
+                        }
                         break;
                      }
                      
+                     // If we are still in the race for flux rope...
+                     if(cellBWConnection[n] < TracingLineEndType::N_TYPES) {
+                        creal distance = sqrt(
+                             (x[0]-(cellInitialCoordinates[n])[0])*(x[0]-(cellInitialCoordinates[n])[0])
+                           + (x[1]-(cellInitialCoordinates[n])[1])*(x[1]-(cellInitialCoordinates[n])[1])
+                           + (x[2]-(cellInitialCoordinates[n])[2])*(x[2]-(cellInitialCoordinates[n])[2])
+                        );
+                        // ...and if we traced too far from the seed, this is not a flux rope candidate and we do a single +=
+                        if(distance > fieldTracingParameters.fluxrope_max_curvature_radii_extent*cellCurvatureRadius[n]
+                           || cellBWRunningDistance[n] > fieldTracingParameters.fluxrope_max_m_to_trace
+                        ) {
+                           cellBWConnection[n] += TracingLineEndType::N_TYPES;
+                        } else if(cellBWRunningDistance[n] > fieldTracingParameters.fluxrope_max_curvature_radii_to_trace*cellCurvatureRadius[n]) {
+                           // If we're still in the game and reach this limit we have a hit and we do a double +=
+                           cellBWConnection[n] += 2*TracingLineEndType::N_TYPES;
```

Comment:
Could you please add some comment description of how this connection type counting is supposed to work?